src/javascript/_common/url.js

```diff
@@ -72,6 +72,12 @@ const Url = (() => {
         return static_host + path.replace(/(^\/)/g, '');
     };
 
+    const setQueryStringWithoutReload = (new_query_str) => {
+        const url = new URL(window.location);
+        url.search = new_query_str;
+        window.history.replaceState('', '', url.href);
```

Comment:
I think we're doing this in other places too, would be good to use this function instead everywhere